pkg/security/authenticators/basic_authentication.go

```diff
@@ -57,7 +70,77 @@ func (a *Basic) Authenticate(request *web.Request) (*web.UserContext, httpsec.De
 		return nil, httpsec.Deny, fmt.Errorf("provided credentials are invalid")
 	}
 
-	bytes, err := json.Marshal(platform)
+	return buildResponse(username, platform)
+}
+
+//BasicOSBAuthenticator attempts to authenticate basic auth requests with provided broker platform credentials
+func BasicOSBAuthenticator(request *web.Request, repository storage.Repository, username, password string) (*web.UserContext, httpsec.Decision, error) {
+	ctx := request.Context()
+	log.C(ctx).Debugf("Attempting to authenticate broker platform credentials")
+
+	brokerID, ok := request.PathParams[osb.BrokerIDPathParam]
+	if !ok {
+		return nil, httpsec.Abstain, fmt.Errorf("could not get authenticate OSB request: brokerID path parameter not found")
+	}
+
+	byBrokerID := query.ByField(query.EqualsOperator, "broker_id", brokerID)
+	byUsername := query.ByField(query.EqualsOperator, "username", username)
+
+	credentialsList, err := repository.List(ctx, types.BrokerPlatformCredentialType, byBrokerID, byUsername)
+	if err != nil {
+		return nil, httpsec.Abstain, fmt.Errorf("could not get credentials entity from storage: %s", err)
+	}
+
+	useOldCredentials := false
+	if credentialsList.Len() != 1 {
+		log.C(ctx).Debugf("Authenticating broker platform credentials failed - will try with to find old credentials")
+
+		byUsername.LeftOp = "old_username"
+		credentialsList, err = repository.List(ctx, types.BrokerPlatformCredentialType, byBrokerID, byUsername)
+		if err != nil {
+			return nil, httpsec.Abstain, fmt.Errorf("could not get credentials entity from storage: %s", err)
+		}
+
+		if credentialsList.Len() != 1 {
+			log.C(ctx).Debugf("Authenticating broker platform credentials failed - will try to fallback to platform credentials authentication")
+			return BasicPlatformAuthenticator(request, repository, username, password)
```

Comment:
this should probably go away 

I think platform credentials should be written as old_credentials/new_credentials initially (probably with migration?) and once they fall out due to rotations, fall backing to them should no longer work